### DISCUSSION TITLE
"labgrid-client -p Place power get" does not find specified usb port

### DISCUSSION THREAD

**Discussion by BenM-github** [labgrid-project/labgrid#1436] [Q&A]
Hi, does anyone know why this error (img below) is occuring? Do I need do add driver information to the exporter or is it my `/etc/udev/rules.d/52-usb.rules` specification? Other commands like `labgrid-client -p PLACE acquire` run without any problem.

**Error:**
![error](https://github.com/labgrid-project/labgrid/assets/79357350/4f935966-2960-4b66-ac84-1b19d02ae03c)
I mean if there is no USBPowerPort I can talk to why can I see them in my labgrid-client resources? This is propably the wrong place for this question but I am unsure where else to find an answer. 

This is what labgrid says about the USBPowerPort **(labgrid-client -p Place show):**
![labgrid-client show](https://github.com/labgrid-project/labgrid/assets/79357350/9e0c27b3-3214-4875-a1d1-1fbb3aaea5ec)

And this is how I present the resource information to the exporter in the **exporter.yaml:**
![Screenshot 2024-06-27 140312](https://github.com/labgrid-project/labgrid/assets/79357350/869a006f-2c16-4480-9509-922cfa24dd0f)

I also read the doc about USBPowerPort (https://labgrid.readthedocs.io/en/latest/configuration.html#usbpowerport) but it seems like my mistake is something different than the USBPowerPort specification.

When I run `uhubctl -l 3-8.4 -p 2` like in the error message I get the return:
`No compatible devices detected at location 3-8.4!
Run with -h to get usage info.`
Makes sence that such error is occuring but why is this resource listed in the first place?

My knowlege of USB-Ports is very limited so I hope there is an answer to this problem at all. 😅 

Thanks and best regard
Ben

**Comment by Emantor** [comment]
The primary problem is that USB Hubs with more than 4 ports are usually built out of two cascaded USB HUB Chips.
And you'll also need to hit the correct point in the chain so that you switch the USB port instead of the connected device.
What does the output of `lsusb -vt` look like? And are you sure that your USB hub is switchable at all? Not every hub support port power switching, so if uhubctl says it isn't switchable your hub might not be compatible.
